src/lib/capacityEngine.js

```diff
@@ -1,186 +1,253 @@
 const { ethers, BigNumber } = require('ethers');
 
-const { MAX_COVER_PERIOD, SECONDS_PER_DAY } = require('./constants');
+const { MAX_COVER_PERIOD } = require('./constants');
 const {
   bnMax,
   calculateTrancheId,
   calculateFirstUsableTrancheIndex,
   calculateProductDataForTranche,
 } = require('./helpers');
-const { selectProduct, selectProductPools } = require('../store/selectors');
+const { selectProduct, selectProductPools, selectProductsInPool } = require('../store/selectors');
 
 const { WeiPerEther, Zero } = ethers.constants;
 
 const BASIS_POINTS = 10000;
 
 /**
- * Calculates the utilization rate of the capacity.
+ * Calculates the index of the first usable tranche for the maximum cover period.
+ * This is used to determine the maximum price a user would get when buying cover.
  *
- * @param {BigNumber} capacityAvailableNXM - The amount of capacity available in NXM.
- * @param {BigNumber} capacityUsedNXM - The amount of capacity used in NXM.
- * @returns {BigNumber} The utilization rate as a BigNumber, expressed in basis points (0-10,000).
- *                      Returns undefined if capacity in NXM is missing.
+ * @param {BigNumber} now - The current timestamp in seconds.
+ * @param {BigNumber} gracePeriod - The product's grace period in seconds.
+ * @returns {number} The index difference between the first usable tranche for max period and the first active tranche.
  */
-function getUtilizationRate(capacityAvailableNXM, capacityUsedNXM) {
-  if (!capacityAvailableNXM || !capacityUsedNXM) {
-    return undefined;
-  }
+function calculateFirstUsableTrancheIndexForMaxPeriod(now, gracePeriod) {
+  const firstActiveTrancheId = calculateTrancheId(now);
+  const firstUsableTrancheIdForMaxPeriod = calculateTrancheId(now.add(MAX_COVER_PERIOD).add(gracePeriod));
+  return firstUsableTrancheIdForMaxPeriod - firstActiveTrancheId;
+}
+
+/**
+ * Calculates the pool-level utilization rate across all products in the pool.
+ *
+ * @param {Array<Object>} products - Array of product capacity data for the pool
+ * @returns {BigNumber} The pool-level utilization rate as a BigNumber, expressed in basis points (0-10,000)
+ */
+function calculatePoolUtilizationRate(products) {
+  let totalCapacityAvailableNXM = Zero;
+  let totalCapacityUsedNXM = Zero;
+
+  products.forEach(product => {
+    totalCapacityAvailableNXM = totalCapacityAvailableNXM.add(
+      product.availableCapacity.find(c => c.assetId === 255)?.amount || Zero,
+    );
+    totalCapacityUsedNXM = totalCapacityUsedNXM.add(product.usedCapacity);
+  });
 
-  const totalCapacity = capacityAvailableNXM.add(capacityUsedNXM);
+  const totalCapacity = totalCapacityAvailableNXM.add(totalCapacityUsedNXM);
   if (totalCapacity.isZero()) {
     return BigNumber.from(0);
   }
 
-  return capacityUsedNXM.mul(BASIS_POINTS).div(totalCapacity);
+  return totalCapacityUsedNXM.mul(BASIS_POINTS).div(totalCapacity);
```

Comment:
This can return over 100% (which is normal after a burn happens). Frontend may or may not treat that scenario and cap to 100%.

Reply:
@mixplore FYI